In [61]:
import pandas as pd

all_ages = pd.read_csv("data/all-ages.csv") 
recent_grads = pd.read_csv("data/recent-grads.csv")

all_ages.head()

Major_code                                  Major  \
0        1100                    GENERAL AGRICULTURE   
1        1101  AGRICULTURE PRODUCTION AND MANAGEMENT   
2        1102                 AGRICULTURAL ECONOMICS   
3        1103                        ANIMAL SCIENCES   
4        1104                           FOOD SCIENCE   

                    Major_category   Total  Employed  \
0  Agriculture & Natural Resources  128148     90245   
1  Agriculture & Natural Resources   95326     76865   
2  Agriculture & Natural Resources   33955     26321   
3  Agriculture & Natural Resources  103549     81177   
4  Agriculture & Natural Resources   24280     17281   

   Employed_full_time_year_round  Unemployed  Unemployment_rate  Median  \
0                          74078        2423           0.026147   50000   
1                          64240        2266           0.028636   54000   
2                          22810         821           0.030248   63000   
3                          64937        3619           0.042679   46000   
4                          12722         894           0.049188   62000   

   P25th    P75th  
0  34000  80000.0  
1  36000  80000.0  
2  40000  98000.0  
3  30000  72000.0  
4  38500  90000.0

In [42]:
all_ages.shape

(173, 11)

In [60]:
recent_grads.head()

Rank  Major_code                                      Major Major_category  \
0     1        2419                      PETROLEUM ENGINEERING    Engineering   
1     2        2416             MINING AND MINERAL ENGINEERING    Engineering   
2     3        2415                  METALLURGICAL ENGINEERING    Engineering   
3     4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING    Engineering   
4     5        2405                       CHEMICAL ENGINEERING    Engineering   

   Total  Sample_size    Men  Women  ShareWomen  Employed      ...        \
0   2339           36   2057    282    0.120564      1976      ...         
1    756            7    679     77    0.101852       640      ...         
2    856            3    725    131    0.153037       648      ...         
3   1258           16   1123    135    0.107313       758      ...         
4  32260          289  21239  11021    0.341631     25694      ...         

   Part_time  Full_time_year_round  Unemployed  Unemployment_rate  Median  \
0        270                  1207          37           0.018381  110000   
1        170                   388          85           0.117241   75000   
2        133                   340          16           0.024096   73000   
3        150                   692          40           0.050125   70000   
4       5180                 16697        1672           0.061098   65000   

   P25th   P75th  College_jobs  Non_college_jobs  Low_wage_jobs  
0  95000  125000          1534               364            193  
1  55000   90000           350               257             50  
2  50000  105000           456               176              0  
3  43000   80000           529               102              0  
4  50000   75000         18314              4440            972  

[5 rows x 21 columns]

In [41]:
recent_grads.shape

(173, 21)

In [57]:
# All values for Major_category
print(all_ages['Major_category'].value_counts())
print(recent_grads['Major_category'].value_counts())


Engineering                            29
Education                              16
Humanities & Liberal Arts              15
Biology & Life Science                 14
Business                               13
Health                                 12
Computers & Mathematics                11
Agriculture & Natural Resources        10
Physical Sciences                      10
Psychology & Social Work                9
Social Science                          9
Arts                                    8
Industrial Arts & Consumer Services     7
Law & Public Policy                     5
Communications & Journalism             4
Interdisciplinary                       1
Name: Major_category, dtype: int64
Engineering                            29
Education                              16
Humanities & Liberal Arts              15
Biology & Life Science                 14
Business                               13
Health                                 12
Computers & Mathematics                11

In [72]:
# All values for Major_category
print(all_ages['Major_category'].value_counts().index)

all_ages_major_categories = dict()
recent_grads_major_categories = dict()

def count_cat(df, col_to_count):
    counts_dictionary = dict()
    cat_index = df[col_to_count].value_counts().index
    for c in cat_index:
        df_count = df[df[col_to_count] == c]
        total = df_count["Total"].sum()
        counts_dictionary[c] = total
    return counts_dictionary

all_ages_major_categories = count_cat(all_ages, 'Major_category')
recent_grads_major_categories = count_cat(recent_grads, 'Major_category')

#recent_grads_major_categories = recent_grads['Major_category'].value_counts().to_dict()
print(recent_grads_major_categories)

#all_ages_major_categories = all_ages['Major_category'].value_counts().to_dict()
print(all_ages_major_categories)

Index(['Engineering', 'Education', 'Humanities & Liberal Arts',
       'Biology & Life Science', 'Business', 'Health',
       'Computers & Mathematics', 'Agriculture & Natural Resources',
       'Physical Sciences', 'Psychology & Social Work', 'Social Science',
       'Arts', 'Industrial Arts & Consumer Services', 'Law & Public Policy',
       'Communications & Journalism', 'Interdisciplinary'],
      dtype='object')
{'Arts': 357130, 'Social Science': 529966, 'Law & Public Policy': 179107, 'Communications & Journalism': 392601, 'Computers & Mathematics': 299008, 'Humanities & Liberal Arts': 713468, 'Interdisciplinary': 12296, 'Physical Sciences': 185479, 'Business': 1302376, 'Biology & Life Science': 453862, 'Industrial Arts & Consumer Services': 229792, 'Engineering': 537583, 'Psychology & Social Work': 481007, 'Education': 559129, 'Health': 463230, 'Agriculture & Natural Resources': 79981}
{'Arts': 1805865, 'Social Science': 2654125, 'Law & Public Policy': 902926, 'Communications & J

Use the 'Low_wage_jobs' and 'Total columns' to calculate the proportion of recent college graduates that worked low wage jobs.
Store the resulting float as `low_wage_percent`.

In [75]:
low_wage_percent = 0.0
low_wage_percent = recent_grads["Low_wage_jobs"].sum() / recent_grads['Total'].sum()
low_wage_percent

0.09852546076122913

Both all_ages and recent_grads datasets have 173 rows, corresponding to the 173 college major codes. This enables us to do some comparisons between the two datasets and perform some initial calculations to see how similar or different the statistics of recent college graduates are from those of the entire population.

Let's now calculate the number of majors where recent grads did better than the overall population.

## for each major
- Increment recent_grads_lower_unemp_count if Unemployment_rate is lower for recent_grads.
- Increment all_ages_lower_unemp_count if Unemployment_rate is lower for all_ages.
- Do nothing if Unemployment_rate is the same for both.

In [116]:
# All majors, common to both DataFrames
majors = recent_grads['Major'].value_counts().index
#print(majors)
recent_grads_lower_unemp_count = 0
all_ages_lower_unemp_count = 0

for m in majors:
    #geting the row for each major, for recent and all ages
    recent_m = recent_grads[recent_grads['Major'] == m]
    all_m = all_ages[all_ages['Major'] == m]
    #print(recent_m)
    
    #extracting the value for unemployment rate
    recent_unemp = recent_m['Unemployment_rate'].values[0]
    all_unemp = all_m['Unemployment_rate'].values[0]
    #print('recent', recent_unemp, type(recent_unemp))
    
    #comparing unemployment rate and increasing the right counter
    if recent_unemp < all_unemp:
        recent_grads_lower_unemp_count += 1
    elif all_unemp < recent_unemp:
        all_ages_lower_unemp_count += 1

print(recent_grads_lower_unemp_count, all_ages_lower_unemp_count)
print(recent_grads_lower_unemp_count + all_ages_lower_unemp_count)

43 128
171
